go/vt/binlog/eventtoken/compare.go

```diff
@@ -32,17 +30,47 @@ func Minimum(ev1, ev2 *querypb.EventToken) *querypb.EventToken {
 	}
 }
 
-// Fresher returns true if ev1 is fresher than ev2. In case of doubt,
-// it returns false.
-//
-// FIXME(alainjobart): if both events come from the same shard, we can
-// compare replication position (after parsing it). Need to implement that.
-// FIMXE(alainjobart): when it's more than just a simple compare, add
-// unit tests too.
-func Fresher(ev1, ev2 *querypb.EventToken) bool {
+// Fresher compares two event tokens.  It returns a negative number if
+// ev1<ev2, zero if they're equal, and a positive number if
+// ev1>ev2. In case of doubt (we don't have enough information to know
+// for sure), it returns -1.
```

Comment:
I'm not too happy with this corner case of the API. I was thinking of returning an error too (for can't compare, or can't parse the positions). The the user would be:

if cmp, err := eventtoken.Compare(ev1, ev2); err != nil || cmp <=0 {
   ...  // can't know, or older value
}

Maybe that would be simpler... If you guys want that API, I'll change it.